airflow/models/dagbag.py

```diff
@@ -195,6 +200,11 @@ def get_dag(self, dag_id, session: Session = None):
                     session=session,
                 )
                 if sd_last_updated_datetime and sd_last_updated_datetime > self.dags_last_fetched[dag_id]:
+                    self.dags = {  # Remove associated dags to re-add them.
+                        key: dag
+                        for key, dag in self.dags.items()
+                        if dag_id not in (key, dag.parent_dag.dag_id)
```

Comment:
Before any suggests `key != dag_id and dag.parent_dag.dag_id != dag_id`, which is both more logically easier to understand and probably more performant (shortcircuiting): pylint forces me to write this, and I can’t figure out a way to turn it off.